pkg/actors/actors.go

```diff
@@ -538,11 +569,71 @@ func (a *actorsRuntime) callRemoteActor(
 	return invokeResponse, nil
 }
 
+func (a *actorsRuntime) callInternalActor(ctx context.Context, req *invokev1.InvokeMethodRequest) (*invokev1.InvokeMethodResponse, error) {
+	actorTypeID := req.Actor()
+	act := a.getOrCreateActor(actorTypeID.GetActorType(), actorTypeID.GetActorId())
+	factory, ok := a.internalActors[act.actorType]
+	if !ok {
+		return nil, fmt.Errorf("internal actor type '%s' is not registered", act.actorType)
+	}
+
+	// The factory is expected to manage its own internal caching of active actor instances
+	impl := factory(a)
+
+	// Internal actors don't support any form of reentrancy (for now)
+	reentrancyID := uuid.New().String()
```

Comment:
Reentrancy is a stable feature for Actors. We should track that as a work item in the project board.

Reply:
Note that in the latest iteration based on `AppChannel`, we're reusing all the existing reentrancy machinery, which means internal actors can also leverage this feature (if they want).